sparse_ho/models.py

```diff
@@ -230,26 +228,29 @@ def _use_estimator(self, X, y, alpha, tol, max_iter):
         dense = self.estimator.coef_[mask]
         return mask, dense, None
 
-    def reduce_X(self, mask):
-        return self.X[:, mask]
+    @staticmethod
+    def reduce_X(X, mask):
+        return X[:, mask]
 
-    def reduce_y(self, mask):
-        return self.y
+    @staticmethod
+    def reduce_y(y, mask):
+        return y
 
     def sign(self, x, log_alpha):
         return np.sign(x)
 
-    def get_primal(self, mask, dense):
+    def get_primal(self, X, y, mask, dense):
         return mask, dense
```

Comment:
this is weird

Reply:
I know, this is because we also take into the svm, once this PR is merged I will open a new one to either improve or remove the svm

Reply:
the name could be more explicit, get_primal_vector ? get_coefs ?

Reply:
coefs are called beta in the API so get_beta ?